lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -348,6 +349,27 @@ def keyspace(self, query: Query) -> JSONOBJECT:
                     self.accumulate(aggregate, m.key, m.value)
         return aggregate
 
+    def daterange(self, request: Request, json: JSONOBJECT, query: Query) -> JSONOBJECT:
+        """Return only the date range of the selected datasets.
+
+        Replace the selected "entities" (normally Dataset columns) with the
+        SQL min and max functions on the dataset upload timestamp so that the
+        generated SQL query will return a tuple of those two values.
+
+        Args:
+            request: The HTTP Request object
+            json: The JSON query parameters
```

Comment:
These two arguments are not referenced in the function...should we note that in some way?  (IIRC, you don't particularly like using `_` as a parameter name...perhaps prefix the parameter names with an underscore, or, at least, add an annotation to their argument descriptions?)

Reply:
I don't like using `_` because you lose the documentation on how the method is *called* without searching up the call tree. Yes, I perhaps unnecessarily kept all three of the helpers consistent, but I like consistency and it doesn't present any real issues. (And, e.g., I could have created a dispatcher binding `helper` to one of the three and then make a single call: if there were four instead of three, I would have, and I considered it briefly at three but decided not to.)

I don't see how it's helpful to try to document that they're "not used" as this means nothing unless we're trying to encourage callers to `None` them out, which I don't like. Again, this is interface documentation and we're documenting the parameters, not how they're currently used internally.

Reply:
It helps ease the dissonance in the reader/reviewer when they see the parameters in the function signature and find that they aren't used in the function.  I figured that that you were trying to keep the interface consistent with that of other functions, which is why I didn't suggest removing them, but it would be good to have some acknowledgement that, _in the particular case of **this** function_, those parameters are not used, _because_ they are part of a common interface.

Reply:
In fact, I noticed that only one of the other helpers used the whole set, so they're already inconsistent and there's no point in passing the additional parameters. Oops. I just cloned the last helper signature without really thinking about it. 😟